From 58a3fa5ffe1785d05744d881c51d0c34583b96e2 Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jdapena@igalia.com>
Date: Thu, 16 Jul 2020 20:46:49 +0200
Subject: Remove explicit keyword in AtomicReference constructor

It breaks compilation in GCC:
../../../chromium-86.0.4209.2/third_party/nearby/src/cpp/core_v2/internal/payload_manager.cc: In constructor 'location::nearby::connections::PayloadManager::PendingPayload::PendingPayload(std::unique_ptr<location::nearby::connections::InternalPayload>, const EndpointIds&,
 bool)':
../../../chromium-86.0.4209.2/third_party/nearby/src/cpp/core_v2/internal/payload_manager.cc:950:28: error: converting to 'location::nearby::AtomicReference<location::nearby::connections::PayloadManager::EndpointInfo::Status>' from initializer list would use explicit constructor 'location::nearby::AtomicReference<T, typename std::enable_if<((sizeof (T) <= sizeof (unsigned int)) && std::is_trivially_copyable<_Tp>::value), void>::type>::AtomicReference(T) [with T = location::nearby::connections::PayloadManager::EndpointInfo::Status; typename std::enable_if<((sizeof (T) <= sizeof (unsigned int)) && std::is_trivially_copyable<_Tp>::value), void>::type = void]'
---
 cpp/platform_v2/public/atomic_reference.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cpp/platform_v2/public/atomic_reference.h b/cpp/platform_v2/public/atomic_reference.h
index 5742724..bbb8c01 100644
--- a/cpp/platform_v2/public/atomic_reference.h
+++ b/cpp/platform_v2/public/atomic_reference.h
@@ -37,7 +37,7 @@ class AtomicReference<T, std::enable_if_t<sizeof(T) <= sizeof(std::uint32_t) &&
     final {
  public:
   using Platform = api::ImplementationPlatform;
-  explicit AtomicReference(T value)
+  AtomicReference(T value)
       : impl_(Platform::CreateAtomicUint32(static_cast<std::uint32_t>(value))) {
   }
   ~AtomicReference() = default;
-- 
2.25.1

